### PR TITLE
Log the registered template and body parser engines

### PR DESCRIPTION
**Pull request by gitblit**
It would be helpful to log the registered template engines with their content types and the bodyparser engines with their accept types.

### CODE REVIEW

**Comment by raphaelbauer** [comment]
Thanks gitblit :) Why don't we just log everything in the constructors of BodyParserEngineManagerImpl and TemplateEngineManagerImpl? That way we don't have to extend the public interface of both classes and we don't have to change NinjaBootstrap... no critique... just a question - logging that information is certainly very useful :)

**Comment by gitblit** [comment]
I like rhythm and pattern.  Moving the logging to the injected constructors could make the pattern of startup logging non-deterministic as you are subject to Guice's internal ordering behavior.  I'll give it a go and report back.

**Comment by raphaelbauer** [comment]
That's certainly a good reason doing logging that way... Let me know what you find...

File: ninja-core/src/main/java/ninja/template/TemplateEngineManagerImpl.java [hunk 1]
```diff
@@ -71,6 +73,11 @@ public TemplateEngineManagerImpl(Provider<TemplateEngineFreemarker> templateEngi
     }
 
     @Override
+    public Set<String> getContentTypes() {
+        return new TreeSet<String>(contentTypeToTemplateEngineMap.keySet());
```
Comment (raphaelbauer) [inline ninja-core/src/main/java/ninja/template/TemplateEngineManagerImpl.java]: Using something like ImmutableSet.copyOf(contentTypeToTemplateEngineMap.keySet()); would make sense imho...

File: ninja-core/src/main/java/ninja/template/TemplateEngineManager.java [hunk 2]
```diff
@@ -27,9 +28,11 @@
 @ImplementedBy(TemplateEngineManagerImpl.class)
 public interface TemplateEngineManager {
 
+    Set<String> getContentTypes();
```
Comment (raphaelbauer) [inline ninja-core/src/main/java/ninja/template/TemplateEngineManager.java:36]: javadoc would be great...

File: ninja-core/src/main/java/ninja/bodyparser/BodyParserEngineManager.java [hunk 3]
```diff
@@ -16,11 +16,15 @@
...
 package ninja.bodyparser;
 
+import java.util.Set;
+
 import com.google.inject.ImplementedBy;
 
 @ImplementedBy(BodyParserEngineManagerImpl.class)
 public interface BodyParserEngineManager {
 
+    Set<String> getContentTypes();
```
Comment (raphaelbauer) [inline ninja-core/src/main/java/ninja/bodyparser/BodyParserEngineManager.java:31]: javadoc would be great :)

File: ninja-core/src/main/java/ninja/bodyparser/BodyParserEngineManagerImpl.java [hunk 4]
```diff
@@ -71,6 +73,11 @@ public BodyParserEngineManagerImpl(Provider<BodyParserEnginePost> bodyParserEngi
     }
 
     @Override
+    public Set<String> getContentTypes() {
+        return new TreeSet<String>(contentTypeToBodyParserMap.keySet());
```
Comment (raphaelbauer) [inline ninja-core/src/main/java/ninja/bodyparser/BodyParserEngineManagerImpl.java]: Using something like ImmutableSet.copyOf(contentTypeToTemplateEngineMap.keySet()); would make sense imho...

File: ninja-core/src/main/java/ninja/template/TemplateEngineManagerImpl.java (hunk 1 above)
Comment (gitblit) [inline ninja-core/src/main/java/ninja/template/TemplateEngineManagerImpl.java]: I'll change it to a sorted immutable list.  I wanted the types deterministically ordered too.  :)

**Comment by raphaelbauer** [comment]
I rather say that one time too often - but whenever I add something to pull requests (comments, questions etc pp) it's NOT meant to be condescending. I only try to keep the whole codebase in good shape and keep certain quality criteria up :) You certainly do a very very good job! And all I say via comments etc may be wrong...

**Comment by raphaelbauer** [comment]
I think it would also be great to have simple unit tests for both getContentTypes implementations.

**Comment by gitblit** [comment]
Where to log.  I'm on the fence here. Which ever you prefer is fine with me.  You have precedent for startup logging like MessagesImpl which complains that I don't have a de translation.  Tell me your preference and I'll make it so.  If it is to move, then perhaps Route logging should be in the RouterImpl to be consistent.

**Comment by raphaelbauer** [comment]
I'd certainly prefer logging inside the constructors. If that turns out to be a really bad idea we can change that to extending the interfaces and logging at NinjaBootstrap...

**Comment by gitblit** [comment]
You want the Logger injected (Ninja) or the Logger instantiated (*Impl) ?

**Comment by raphaelbauer** [comment]
instantiated via LoggerFactory.getLogger... There is not yet a way how tell the injected logger in which class it logs...

**Comment by gitblit** [comment]
I'll move the Routing table to be consistent, but I'll leave *.getContentTypes() - with all review comments satisfied - since there might be other uses for that info.

**Comment by gitblit** [comment]
Changes pushed.

**Comment by raphaelbauer** [comment]
very nice. Will merge locally... Thanks!

**Pull request merged**